### DISCUSSION TITLE
Native machine code (.mpy) for ESP32

### DISCUSSION THREAD

**Discussion by Gurkengewuerz** [micropython/micropython#10297] [Libraries & Drivers] [answered]
Hello everyone,

i have created a C Module for my ESP32(-S3) to use it in promiscuous mode. It successfully compiles as C Module.
My goal is now to compile it to native machine code (.mpy). I know there are some cons doing so but i need this as a dynamic module.

Looking at the `examples/natmods/` there are just simple examples. I tried adding CFlags for the `esp-idf` components but that is almost impossible.
I know `esp-idf` is using CMake. Are there any examples how i can use the `esp-idf` in native code modules? Or maybe how i can use CMake for the native modules?

Greetings from Germany

**Comment by dpgeorge** [comment, marked as answer]
At the moment it's not possible to create native .mpy files that access the ESP IDF.  It's very complicated because the ESP IDF exposes a lot of symbols and they would need to be known by the MicroPython firmware so it could link them when importing the native .mpy.

**Reply by Gurkengewuerz** [reply]
Hey @dpgeorge thanks for your answer. I'm curious what you mean with "At the moment"? Are there any plans? 👀 
And btw happy Holidays :)

**Reply by jimmo** [reply]
This is not currently something that is being worked on. It is potentially a lot of work and will add a firmware size cost to all users, even though it's possibly something that very few will use.
